From: Andrei Borzenkov <arvidjaar@gmail.com>
Subject: Also skip legacy grub if grub2 is found

Check for grub.cfg also in /boot/grub2, not only in /boot/grub.
Index: os-prober/linux-boot-probes/mounted/x86/40grub
===================================================================
--- os-prober.orig/linux-boot-probes/mounted/x86/40grub
+++ os-prober/linux-boot-probes/mounted/x86/40grub
@@ -95,7 +95,9 @@ fi
 
 if [ "$grubconf" ] && \
    ([ ! -e "$mpoint/boot/grub/grub.cfg" ] || \
-    [ "$mpoint/boot/grub/$grubconf" -nt "$mpoint/boot/grub/grub.cfg" ]); then
+    [ "$mpoint/boot/grub/$grubconf" -nt "$mpoint/boot/grub/grub.cfg" ]) && \
+   ([ ! -e "$mpoint/boot/grub2/grub.cfg" ] || \
+    [ "$mpoint/boot/grub/$grubconf" -nt "$mpoint/boot/grub2/grub.cfg" ]); then
 	debug "parsing $grubconf"
 	parse_grub_menu "$mpoint" "$partition" "$bootpart" < "$mpoint/boot/grub/$grubconf"
 fi
